Vagrantfile.openstack

```diff
@@ -0,0 +1,177 @@
+      vm.vm.box = "dummy"
+      vm.vm.hostname = "#{hostname}.#{munge_domainname(OS_TENANTNAME)}.#{OS_ENDPOINT}"
+      vm.vm.provider :openstack do |os|
+       vm_settings.call(os)
+       os.server_name = hostname
+       os.floating_ip = FLOATING_IPS[num]
+       os.disks = [{"name" => "#{hostname} disk 1", "description" => "Chef-BCPC Disk", "size" => DISK_SIZE},
+                   {"name" => "#{hostname} disk 2", "description" => "Chef-BCPC Disk", "size" => DISK_SIZE}]
+      end
+      vm.vm.provision :shell, :inline => "echo 'Acquire::http::Proxy \"#{proxy_url}\";' >> /etc/apt/apt.conf" if proxy_url
+
+      # Volume attachment is inconsistent on first boot, so reboot before Chefing is tried
+      vm.vm.provision :shell, :inline => "reboot"
+    end
+  end
+
+  config.vm.define "#{CLUSTER_NAME}bcpcchef".to_sym, primary: true, autostart: false  do |bcpcchef|
+    hostname="#{CLUSTER_NAME}bcpc-chef"
+    bcpcchef.ssh.private_key_path = "vagrant_ssh/id_rsa"
+
+    openstack_compute = Fog::Compute.new({:provider           => :openstack,
+                                          :connection_options => { :ssl_verify_peer => false },
+                                          :openstack_username => ENV["OS_USERNAME"],
+                                          :openstack_api_key  => ENV["OS_PASSWORD"],
+                                          :openstack_auth_url => "https://#{OS_ENDPOINT}:5000/v2.0/tokens",
+                                          :openstack_tenant   => OS_TENANTNAME
+                                          })
+
+    machines = openstack_compute.servers.all( :name => "#{CLUSTER_NAME}bcpc-1" )
+    id = machines.first.id if machines.length() == 1
+    # Set a bogus address incase we do not have a bcpc-1 VM at this time
+    vm1_fixed_ip = "## DID NOT FIND #{CLUSTER_NAME}bcpc-1 VM ##"
+    if not id
+      nil
+      puts "Unable to find host #{CLUSTER_NAME}bcpc-1 found #{machines.length()} matches"
+    else
+      ips = openstack_compute.servers.get(id).private_ip_addresses
+      if not ips.length() == 1
+        nil
+        puts "Unable to find IP for #{CLUSTER_NAME}bcpc-#{idx} found #{ips.length()} IPs"
+      end
+      vm1_fixed_ip = ips.first
+    end
+
+    # Every Vagrant virtual environment requires a box to build off of.
+    bcpcchef.vm.box = "dummy"
+    fqdn = "#{hostname}.#{munge_domainname(OS_TENANTNAME)}.#{OS_ENDPOINT}"
+    raise ArgumentError, "hostnames longer than 64 characters not permitted per https://docs.getchef.com/server_security.html\nTrying: #{fqdn}" if fqdn.length > 64
+    bcpcchef.vm.hostname = fqdn
+    bcpcchef.vm.provider :openstack do |os|
+     vm_settings.call(os)
+     os.server_name = hostname
+     os.floating_ip = FLOATING_IPS[0]
+     os.flavor = /m1.medium/
+    end
+
+    # spoof having a vagrant homedir
+    bcpcchef.vm.provision :shell, :inline => "ln -s /home/ubuntu /home/vagrant ; ln -s /vagrant /home/ubuntu/chef-bcpc ; chown -R ubuntu /home/ubuntu/" 
```

Comment:
This and following vm.provision statements would read better in a ruby-block-style provision block.